move/abi/sources/abi.move

```diff
@@ -133,33 +127,36 @@ public fun read_vector_bytes(self: &mut AbiReader): vector<vector<u8>> {
     let length = self.read_u256() as u64;
     self.head = self.pos;
 
-    let mut i = 0;
-
-    while (i < length) {
-        var.push_back(self.read_bytes());
-
-        i = i + 1;
-    };
+    length.do!(|_| var.push_back(self.read_bytes()));
 
     // Move position to the next slot
-    self.pos = pos + 32;
+    self.pos = pos + BYTES_IN_U256;
     self.head = head;
 
     var
 }
 
+public fun read_bytes_raw(self: &mut AbiReader): vector<u8> {
+    // Move position to the start of the bytes
+    let offset = self.read_u256() as u64;
+    let length = self.bytes.length() - offset;
```

Comment:
This is reading the full bytes until the end. What if you there are multiple encoded structs?

Reply:
if it's only meant to support a single one, then it should consume the reader so it can't be used again

maybe it can return the new reader itself

Reply:
It is not supposed to include only one, but there is no way for us to know up to what point the raw bytes are actually the struct the user needs. For example, assume someone encodes a `struct` {var1:`u256`, var2:`u256`} followed by some data: `bytes`. The encoded data will look like this: `[offset1 = 64, offset2=128, var1, var2, data.length, data]`. There is no way to know where the `struct` ends and where the `data` begins. If however the struct is the last variable length variable encoded, then it will take up all of the remaining bytes. This is why we return everything and expect the user to decode only as much as they need (knowing the format of the data is required, and it will tell the user where to stop.)